QualityControl/lib/database/SequelizeDatabase.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file 'COPYING'.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { LogManager } from '@aliceo2/web-ui';
+import { Sequelize } from 'sequelize';
+import pkg from 'umzug';
+import { dirname, join } from 'path';
+import { fileURLToPath } from 'url';
+import { createUmzug } from './umzug.js';
+import { getDbConfig } from '../config/database.js';
+import models from './models/index.js';
+const { SequelizeStorage } = pkg;
+
+/**
+ * Sequelize implementation of the Database.
+ */
+export class SequelizeDatabase {
+  constructor(config) {
+    this._logger = LogManager.getLogger('qcg/database');
+
+    if (!config) {
+      this._logger.warnMessage('No configuration provided for SequelizeDatabase. Using default configuration.');
+    }
+    this.dbConfig = getDbConfig(config);
+    const {
+      database,
+      username,
+      password,
+      host,
+      port,
+      charset,
+      collate,
+      timezone,
+      logging,
+    } = this.dbConfig;
+
+    const __filename = fileURLToPath(import.meta.url);
+    this.__dirname = dirname(__filename);
```

Comment:
Any reason why this is defined in the constructor if it is used only in the migrate method?